vncviewer/ServerDialog.cxx

```diff
@@ -53,6 +54,23 @@ static LogWriter vlog("ServerDialog");
 
 const char* SERVER_HISTORY="tigervnc.history";
 
+static char* usedDir = NULL;
+static void updateUsedDir(const char* filename)
+{
+#ifndef  _WIN32
+  char * name = strdup(filename);
+  usedDir = strdup(dirname(name));
+  free(name);
+#else
+  char dir[_MAX_DIR], drive[_MAX_DRIVE], path[_MAX_PATH];
+  if (_splitpath_s(filename, drive, _MAX_DRIVE, dir, _MAX_DIR, NULL, 0, NULL, 0) == 0) {
+    if (_makepath_s( path, _MAX_PATH, drive, dir, NULL, NULL) == 0) {
```

Comment:
`dirname()` seems to be provided by mingw, so I don't think we need this alternative for Windows?

Reply:
Yes, you are right, it is working. Will remove it.